datasets/newsgroup/newsgroup.py

```diff
@@ -0,0 +1,171 @@
+# coding=utf-8
+# Copyright 2020 The TensorFlow Datasets Authors and the HuggingFace NLP Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Lint as: python3
+"""20Newsgroup  dataset"""
+
+from __future__ import absolute_import, division, print_function
+
+import csv
+import os
+
+import nlp
+
+
+_CITATION = """
+    @inproceedings{Lang95
+    ,author = "Ken Lang"
+    ,title = "Newsweeder: Learning to filter netnews"
+    ,year = 1995
+    ,booktitle = "Proceedings of the Twelfth International Conference on Machine Learning"
+    ,pages = "331-339"
+    }
+ """
+
+_DESCRIPTION = """
+The 20 Newsgroups data set is a collection of approximately 20,000 newsgroup documents, partitioned (nearly) evenly across 
+20 different newsgroups. To the best of my knowledge, it was originally collected by Ken Lang, probably for his Newsweeder: 
```

Comment:
whose "knowledge" ? ^^
Is there a description that is more factual ?

Reply:
You're right.